src/Generics/SOP/Arbitrary.hs

```diff
@@ -25,13 +25,41 @@ import Generics.SOP
 --
 -- > instance Arbitrary T where
 -- >   arbitrary = garbitrary
+-- >   shrink    = gshrink
 --
 -- Note that currently no attempts are being made to generate arbitrary
 -- values of a particular size, and it is possible that this function
 -- diverges for recursive structures.
 --
 garbitrary :: forall a. (Generic a, All2 Arbitrary (Code a)) => Gen a
-garbitrary = liftM to $ hsequence =<< elements (apInjs_POP $ hcpure p arbitrary)
+garbitrary = fmap to $ oneof (map hsequence $ apInjs_POP $ hcpure p arbitrary)
   where
     p :: Proxy Arbitrary
     p = Proxy
+-- Note: we could generate all elements of sums and pick on with 'elements'.
```

Comment:
I see, I somehow though that we get `SOP Gen (Code a) -> Gen (SOP I (Code a))` with `hsequence`. I'll revert the change and add comments in between.